### ISSUE TITLE
A few questions

### ISSUE THREAD

**Issue by dahanj95** [SteveWinward/GoogleSheetsWrapper#21]
The first one, where's the doc for the library? I was grinding the entire git in order to avoid opening an "issue" but couldn't find, so it leaves me with no choice but ask instead.

- I see a parameter in the **SheetField** annotation called "**NumberFormatPattern**" which is helpful, but how can I format the string to include the value itself? For example, if I have the property of type int "SomeValue" and it is annotated with **SheetField**, I want to use the value of this property to format the **NumberFormatPatter**, something like string interpolation introduced in C# 6.0 $"Value is {SomeValue}" but wouldn't let me.

- How do I write a Header to a newly created sheet that has no Header? I know there's a property "**HasHeaderRow**" which is helpful, but I want to use it to write a Header once to a new sheet. It would be better if the helper library would do it for me instead of having a nasty if statement.

**Comment by SteveWinward** [comment]
@dahanj95,

With regards to the ````SheefFieldAttribute```` class, there is not a way to use interpolation with the ````NumberFormatPattern```` property.  The ````NumberFormatPattern```` property simply lets you define the Google Sheets API defined format property of the cell.  For full details of the various patterns you can use, see the link below.  I'll also add these to the docs as well to make it easier to find in the future.

https://developers.google.com/sheets/api/guides/formats

For writing the header row you have a few options.  I added a new method in the ````BaseRepository```` class just now (version 1.0.52) to include a method ````WriteHeaders```` which will append the header display names to the sheet.  This will throw an exception if the ````HasHeaderRow```` is true.  

````csharp
// You need to implement your own configuration management solution here!
var settings = AppSettings.FromEnvironment();

// Create a SheetHelper class for the specified Google Sheet and Tab name
var sheetHelper = new SheetHelper<TestRecord>(
    settings.GoogleSpreadsheetId,
    settings.GoogleServiceAccountName,
    settings.GoogleMainSheetName);

sheetHelper.Init(settings.JsonCredential);

// Create a Repository for the TestRecord class
var repository = new TestRepository(sheetHelper, hasHeaderRow: false);

// Write the header display names to the Google Sheet tab
repository.WriteHeaders();
````

If you want to ensure the header columns are present, the best way to do that is to make sure you call the ````ValidateSchema```` method which will inspect the first row to make sure it has all the column header values.

Depending on what you are doing, you may also want to check out the ````SheetAppender```` class,

https://github.com/SteveWinward/GoogleSheetsWrapper#append-a-csv-to-google-sheets

Hope this helps.

**Comment by dahanj95** [comment]
Thank you very much for replying back, I have another question. How can I create a new sheet (not document, a sheet within same document) with your library? Thanks again.

**Comment by dahanj95** [comment]
After calling the UpdateTabName to create a new sheet, how do I point an existing SheetHelper instance to the new sheet without having to re-create a new instance?

**Comment by dahanj95** [comment]
Hi Steve, just a reminder that I'm waiting for a response for my last comment. Thanks.

**Comment by SteveWinward** [comment]
@dahanj95 sorry for the delay. I built this library in my free time and I haven’t had a chance to review your latest comment. I suspect you are right. I quickly made a change hoping it would solve your issue I probably did the wrong logic. 

I reopened the issue in case anyone in the community is able to fix this. Once I get more bandwidth I will try and take a look again. 

Sorry you are dealing with this issue.
